components/src/validation/rules.ts

```diff
@@ -304,11 +304,11 @@ export const validPhoneNumberFormat = function(
   );
 };
 
-export const validURLFormat = function(value: string): boolean | string {
+export const validURLFormat = function (value: string): boolean | string {
   return (
     !value ||
-    /^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!10(?:\.\d{1,3}){3})(?!127(?:\.\d{1,3}){3})(?!169\.254(?:\.\d{1,3}){2})(?!192\.168(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/[^\s]*)?$/i.test(
-      value,
```

Comment:
It is better to write a test for the mentioned scenario in the stack overflow. @gayan-orangehrm after KT from Thilina please try to attempt this.

Reply:
Sure, I will do